lib/constants.js

```diff
@@ -1,6 +1,35 @@
+const os = require('os'),
+  process = require('process');
+
 exports.PROTOCOL = {
   TCP: 'tcp',
   UDS: 'uds',
   UDP: 'udp',
   STREAM: 'stream'
 };
+
+/**
+ * Determines error codes that signify a connection to a Unix Domain Socket (UDS)
+ * has failed in a way that can be retried. This codes are OS-specific.
+ * @returns {number[]} An array of the error codes.
+ */
+function udsErrors() {
+  if (process.platform === 'linux') {
+    return [
+      os.constants.errno.ENOTCONN,
+      os.constants.errno.ECONNREFUSED,
```

Comment:
Huh, I didn't realize there was a constant here available on Node 8 and above.  Good to know.